### PR TITLE
Kernel: Be a bit more judicious about how we copy USB structures

### PR DESCRIPTION
**Pull request by Hendiadyoin1**
A lot of USB structures contain back pointers to their parents, so we need to be careful about how we copy them around, and update the back pointers as necessary.

---

Also Fix lsusb

### CODE REVIEW

**Comment by Hendiadyoin1** [comment]
CC: @spholz 
CC: @sdomi

File: Kernel/Bus/USB/USBConfiguration.cpp [hunk 1]
```diff
@@ -14,6 +14,27 @@
 
 namespace Kernel::USB {
 
+USBConfiguration::USBConfiguration(USBConfiguration const& other)
+    : m_device(other.m_device)
+    , m_descriptor(other.m_descriptor)
+    , m_descriptor_index(other.m_descriptor_index)
+    , m_interfaces(other.m_interfaces)
+{
+    // FIXME: This can definitely OOM
```
Comment (sdomi) [inline Kernel/Bus/USB/USBConfiguration.cpp:23]: what would be the fix here, even? should we somehow hardcap the amount of USB devices?

(i don't see this as a blocker, but I would like to look into fixing it some day - i don't want serenity to be pwnable like a PS3 :p)

File: Kernel/Bus/USB/USBConfiguration.h [hunk 2]
```diff
@@ -39,7 +47,7 @@ class USBConfiguration {
     ErrorOr<void> enumerate_interfaces();
 
 private:
-    Device& m_device;                              // Reference to the device linked to this configuration
+    Device* m_device;                              // Reference to the device linked to this configuration
```
Comment (sdomi) [inline Kernel/Bus/USB/USBConfiguration.h:50]: nit: Well Technically Now It's A Pointer, so the comment is invalid-ish

**Review by sdomi — COMMENTED** [review]
good job, lgtm!

File: Kernel/Bus/USB/USBConfiguration.cpp (hunk 1 above)
Comment (Hendiadyoin1) [reply Kernel/Bus/USB/USBConfiguration.cpp:23]: there are two solutions:
1. Avoid this all together
2. Make a fallible

File: Kernel/Bus/USB/USBConfiguration.h (hunk 2 above)
Comment (Hendiadyoin1) [reply Kernel/Bus/USB/USBConfiguration.h:50]: I would want to add a nonnull attribute to that, but that seems to be for function arguments/returns only?
Ideally we'd have a transparent nonnullptr/reference wrapper that allows pointer assignment

**Pull request merged**